meetup/tests.py

```diff
@@ -113,95 +154,70 @@ def test_invalid_url_returns_404(self):
 
   def test_validator_returns_true_for_valid_url(self):
     lookup_agent = self.lookup_agent_builder("http://meetup.com/papers-we-love/")
-    is_valid = lookup_agent.is_real_meetup()
+    is_valid = lookup_agent.url_exists()
     self.assertTrue(is_valid)
 
   def test_validator_returns_false_for_invalid_url(self):
     lookup_agent = self.lookup_agent_builder("http://meetup.com/this-is-not-a-meetup/")
-    is_valid = lookup_agent.is_real_meetup()
+    is_valid = lookup_agent.url_exists()
     self.assertFalse(is_valid)
 
-  def test_events_lookup_returns_event(self):
-    lookup_agent = self.lookup_agent_builder("http://meetup.com/papers-we-love/")
-    response = lookup_agent.get_response('events')
-    self.assertEqual(response.status_code, 200)
-
   def lookup_agent_builder(self, link):
-    return MeetupApiLookupAgent(link)
+    return MeetupApi(link)
+
 
-@patch("meetup.services.meetup_api_lookup_agent.MeetupApiLookupAgent")
-class TestMeetupInfoFetch(TestCase):
+class TestMeetupPresenter(TestCase):
 
-  def create_meetup_with_associated_pizza(self):
-    meetup = Meetup.objects.create(name="Meeetup1", meetup_link='http://www.meetup.com/papers-we-love/')
-    meetup.pizza_places.create(name="PizZap")
-    return meetup
+  def setUp(self):
+    self.mock_meetup_api = MagicMock(spec=MeetupApi)
+    self.mock_meetup_api.return_value.get_response.return_value.json.return_value = meetup_api_response
```

Comment:
This long train is a smell. What if there was another layer between the presenter and API, so you didn't have to reach so deep to configure the API's return values? :steam_locomotive: :train: :train: :train: :train: